src/components/DatePicker/DatePicker.tsx

```diff
@@ -80,10 +80,24 @@ const DatePicker: React.FC<Props> = ({
   const [selectedDay, setSelectedDay] = useState<DateRange>(value);
 
   useEffect(() => {
-    if (onChange) {
-      onChange(selectedDay);
+    if (value.from && value.to) {
+      setSelectedDay(value);
     }
-  }, [onChange, selectedDay]);
+  }, [value]);
+
+  const handleCalendarValueChange = useCallback(
+    (range, hideDatePicker = false) => {
+      setSelectedDay(range);
```

Comment:
Here you can create a stronger cohesion between the updates by adding a second argument (callback fn) which will run after the update of the selectedDay. 

example:
`setSelectedDay(range, () => setSelectedOption('custom')); `

PS: Although, I didn't find a way to reproduce it on typescript as the SetStateAction accepts 1 argument.
```type SetStateAction<S> = S | ((prevState: S) => S);```